docs/content/contributing/tests.md

```diff
@@ -0,0 +1,127 @@
+
+### Structure
+
+The `test/` directory contains multiple directories. Among them is the `bats/` directory (_which is the [BATS] git submodule_) and the `helper/` directory. The latter is especially interesting because it contains common support functionality used in almost every test. Actual tests are located in `test/tests/`.
+
+!!! warning "Test suite undergoing refactoring"
+
+    We are currently in the process of restructuring all of our tests. Tests will be moved into `test/tests/parallel/` and new tests should be placed there as well.
+
+### How Are Tests Run?
+
+Tests are split into two categories:
+
+- **`test/tests/parallel/`:** Multiple test files are run concurrently to reduce the required time to complete the test suite. A test file will presently run it's own defined test-cases in a sequential order.
+- **`test/tests/serial/`:** Each test file is queued up to run sequentially. Tests that are unable to be support being run concurrently belong here.
+
+Parallel tests are further partitioned into smaller sets. If your system has the resources to support running more than one of those sets at a time this is perfectly ok (_our CI runs tests by distributing the sets across multiple test runners_). Care must be taken not to mix running the serial tests while a parallel set is also running; this is handled for you when using `make tests`.
+
+## Running Tests
+
+### Prerequisites
+
+To run the test suite, you will need to:
+
+1. [Install Docker][get-docker]
+2. Install `jq` and (GNU) `parallel` (under Ubuntu, use `sudo apt-get -y install jq parallel`)
+3. Execute `git submodule update --init --recursive` if you haven't already initialized the git submodules
+
+### Executing Test(s)
+
+We use `make` to run commands. You will first need to build the container image via `make build`. You can then:
+
+1. Run all tests: `make clean tests`
+2. Run a single test: `make clean generate-accounts test/<TEST NAME WITHOUT .bats SUFFIX>`
+3. Run multiple unrelated tests: `make clean generate-accounts test/<TEST NAME WITHOUT .bats SUFFIX>,<TEST NAME WITHOUT .bats SUFFIX>` (just add a `,` and then immediately write the new test name)
+4. Run a whole set or all serial tests: `make clean generate-accounts tests/parallel/setX` where `X` is the number of the set or `make clean generate-accounts tests/serial`
+
+??? tip "Setting the Degree of Parallelization for Tests"
+
+    If your machine is capable, you can increase the amount of tests that are run simultaneously by prepending the `make clean all` command with `BATS_PARALLEL_JOBS=X` (i.e. `BATS_PARALLEL_JOBS=X make clean all`). This wil speed up the test procedure. You can also run all tests in serial by setting `BATS_PARALLEL_JOBS=1` this way.
+
+    The default value of `BATS_PARALLEL_JOBS` is 2. If your system has the available resources spare to support increasing the amount of active jobs, the test suite can finish more quickly.
```

Comment:
```suggestion
    The default value of `BATS_PARALLEL_JOBS` is 2. This can be increased if your system has the resources spare to support running more jobs at once to complete the test suite sooner.
```